### PR TITLE
support exclude_filepaths

### PR DESCRIPTION
**Pull request by ghbren**
Add support for exclude_filepaths option.

### CODE REVIEW

File: spec/fixtures/pattern_search/test_paths/salus.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+scanner_configs:
+  PatternSearch:
+    matches:
+      - regex: hello
+        forbidden: true
+        exclude_filepaths:
```
Comment (xolaz) [inline spec/fixtures/pattern_search/test_paths/salus.yaml:6]: can you exclude filepaths and directories at the same time? If so, can you make a test case for it?

Comment (ghbren) [reply spec/fixtures/pattern_search/test_paths/salus.yaml:6]: Yes, just added test case

File: spec/fixtures/pattern_search/test_paths2/salus.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    matches:
+      - regex: hello
+        forbidden: true
+        exclude_extension:
+          - html
+        exclude_filepaths:
+          - file.rb
+          - file.py
+        exclude_directory:
+          - subdir
```
Comment (xolaz) [inline spec/fixtures/pattern_search/test_paths2/salus.yaml:12]: what if the directory is nested, what's the format for that? can you update the readme with that detail at least?

Comment (ghbren) [reply spec/fixtures/pattern_search/test_paths2/salus.yaml:12]: updated the doc for `exclude_directory`

**Pull request merged**